src/schema/v2/me/myCollection.ts

```diff
@@ -121,29 +127,54 @@ export const MyCollection: GraphQLFieldConfig<any, ResolverContext> = {
     // @ts-expect-error FIXME: Make `page` is an optional param of `gravityOptions`
     delete gravityOptions.page
 
-    return collectionArtworksLoader("my-collection", gravityOptions)
-      .then(({ body, headers }) => {
-        return connectionFromArraySlice(body, options, {
-          arrayLength: parseInt(headers["x-total-count"] || "0", 10),
-          sliceStart: gravityOptions.offset,
-        })
-      })
-      .catch((error) => {
-        console.error("[schema/v2/me/my_collection] Error:", error)
-
-        if (error.message == "Collection Not Found") {
-          // For some users with no items, Gravity produces an error of
-          // "Collection Not Found". This can cause the Gravity endpoint to
-          // produce a 404, so we will intercept the error and return an empty
-          // list instead.
-          return connectionFromArray([], options)
-        } else {
-          throw error
-        }
+    try {
+      const { body: artworks, headers } = await collectionArtworksLoader(
```

Comment:
Can we fire both requests in parallel with `Promise.all`? 

```
const [{ body: artworks, headers }, submissions] = await Promise.all([collectionArtworksLoader("my-collection", gravityOptions), submissionsLoader({ size: 1000 })])

Reply:
Or something that is easier to read :)

Reply:
sure, will do!